cmd/nebraska/controller.go

```diff
@@ -821,8 +822,12 @@ func (ctl *controller) getMetrics(c *gin.Context) {
 
 func (ctl *controller) processOmahaRequest(c *gin.Context) {
 	c.Writer.Header().Set("Content-Type", "text/xml")
+	c.Request.Body = http.MaxBytesReader(c.Writer, c.Request.Body, UpdateMaxRequestSize)
 	if err := ctl.omahaHandler.Handle(c.Request.Body, c.Writer, getRequestIP(c.Request)); err != nil {
 		logger.Error("process omaha request", "error", err)
+		if err.Error() == fmt.Errorf("%s: %s", omaha.ErrMalformedRequest, "http: request body too large").Error() {
```

Comment:
Could you see if `if uerr := errors.Unwrap(err); uerr != nil && uerr.Error() == "http: request body too large"` would work?
```suggestion
		if uerr := errors.Unwrap(err); uerr != nil && uerr.Error() == "http: request body too large" {
```